### PR TITLE
Add dsim host

### PR DESCRIPTION
**Pull request by brickZA**
@paulprozesky Have a look!

### CODE REVIEW

File: src/dsimhost_fpga.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            self, host, katcp_port=katcp_port, boffile=boffile, connect=connect)
+        self.boffile = boffile
+        self.sine_sources = AttributeContainer()
+        self.noise_sources = AttributeContainer()
+        self.outputs = AttributeContainer()
+
+    def get_system_information(self, filename=None, fpg_info=None):
+        """Get system information and build D-engine sources"""
+        FpgaHost.get_system_information(self, filename=filename, fpg_info=fpg_info)
+        self.sine_sources.clear()
```
Comment (paulprozesky) [inline src/dsimhost_fpga.py]: AttributeContrainer doesn't have a clear() method - although it probably should. Have you added one?

File: src/dsimhost_fpga.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            if sin_name is not None:
+                setattr(self.sine_sources, 'sin_'+ sin_name, SineSource(reg, sin_name))
+            elif noise_name is not None:
+                setattr(self.noise_sources, 'noise_' + noise_name,
+                        NoiseSource(reg, noise_name))
+            elif output_scale_name is not None:
+                setattr(self.outputs, 'out_' + output_scale_name,
+                        Output(output_scale_name, reg, self.registers.control))
+
+    def initialize(self):
```
Comment (paulprozesky) [inline src/dsimhost_fpga.py]: initialiSe :)

File: src/dsimhost_fpga.py (hunk 1 above)
Comment (brickZA) [inline src/dsimhost_fpga.py]: Oops, see https://github.com/ska-sa/casperfpga/pull/9

File: src/dsimhost_fpga.py (hunk 2 above)
Comment (brickZA) [inline src/dsimhost_fpga.py]: fixed, see push :)

**Pull request merged**